hat/app/org/hatdex/hat/api/controllers/HattersRequestProxy.scala

```diff
@@ -0,0 +1,84 @@
+package org.hatdex.hat.api.controllers
+
+import com.mohiva.play.silhouette.api.repositories.AuthInfoRepository
+import com.mohiva.play.silhouette.api.util.PasswordHasherRegistry
+import com.mohiva.play.silhouette.api.Silhouette
+import javax.inject.Inject
+import org.hatdex.hat.api.json.ApplicationJsonProtocol
+import org.hatdex.hat.api.models._
+import org.hatdex.hat.api.service.{ MailTokenService, RemoteExecutionContext, UsersService }
+import org.hatdex.hat.api.service.applications.ApplicationsService
+import org.hatdex.hat.authentication.{ ContainsApplicationRole, HatApiAuthEnvironment, HatApiController, WithRole }
+import org.hatdex.hat.phata.models.MailTokenUser
+import org.hatdex.hat.api.json.HatJsonFormats._
+import org.hatdex.hat.resourceManagement.HatServer
+import play.api.http.HttpEntity
+import play.api.{ Configuration, Logger }
+import play.api.libs.json.Json
+import play.api.libs.ws.WSClient
+import play.api.mvc.{ Action, AnyContent, ControllerComponents }
+
+import scala.concurrent.Future
+
+class HattersRequestProxy @Inject() (
+    components: ControllerComponents,
+    silhouette: Silhouette[HatApiAuthEnvironment],
+    wsClient: WSClient)(
+    implicit
+    val ec: RemoteExecutionContext,
+    applicationsService: ApplicationsService,
+    tokenService: MailTokenService[MailTokenUser],
+    authInfoRepository: AuthInfoRepository[HatServer],
+    passwordHasherRegistry: PasswordHasherRegistry,
+    usersService: UsersService,
+    configuration: Configuration)
+  extends HatApiController(components, silhouette) with ApplicationJsonProtocol {
+
+  val logger = Logger(this.getClass)
+
+  def proxyRequestHatClaim(claimToken: String): Action[AnyContent] = SecuredAction(ContainsApplicationRole(Owner()) || WithRole(Owner())).async { implicit request =>
```

Comment:
The controller should be further refactored:
1. Secured action should be replaced with UserAwareAction
2. It should accept and validate JSON input send by Rumpel.
3. Should forward JSON input to Hatters endpoint

Reply:
Fixed.